### ISSUE TITLE
0.6.1 release breaks code depending on 0.6.0

### ISSUE THREAD

**Issue by linkmauve** [RustCrypto/hashes#34]
This is a violation of semver, the error being:
```rust
error[E0599]: no method named `variable_result` found for type `blake2::Blake2b` in the current scope                  
   --> src/caps.rs:180:35                                  
    |                                                                                                                  
180 |                 let hash = hasher.variable_result(&mut buf).unwrap();                                            
    |                                   ^^^^^^^^^^^^^^^    
    |                                                                                                                  
    = note: the method `variable_result` exists but the following trait bounds were not satisfied:                     
            `blake2::Blake2b : digest::ExtendableOutput`   
            `&blake2::Blake2b : digest::ExtendableOutput`                                                              
            `&mut blake2::Blake2b : digest::ExtendableOutput`
```

**Comment by newpavlov** [comment]
I am very sorry that you got influenced by this, but it was a conscious decision, partially to notify users of this feature about the [problem](https://github.com/RustCrypto/hashes/issues/28). As it's quite a serious bug, I think it's better to see `0.6.0` as a broken release. I will publish `0.7.0` with the fix hopefully this week. Meanwhile if you want the same (incorrect) behavior you can either truncate fixed result or use `Cargo.lock`.

**Comment by linkmauve** [comment]
Thank you for the reply, in the meantime I released [xmpp-parsers 0.7.1](https://crates.io/crates/xmpp-parsers/0.7.1) with [blake2b disabled](https://hg.linkmauve.fr/xmpp-parsers/rev/779d8c8e1349).  I’ll release a 0.7.2 once you have a fix.

**Comment by newpavlov** [comment]
I've updated `blake2` crate to v0.7. Now it should have a valid implementation of variable output. Note that `VariableOutput` trait signature has [changed](https://docs.rs/digest/0.7.1/digest/trait.VariableOutput.html).
